proto/control-api/src/grpc/convert/api.rs

```diff
@@ -230,55 +255,72 @@ impl TryFrom<proto::apply_request::El> for Room {
     }
 }
 
-impl TryFrom<proto::Room> for Room {
+impl TryFrom<proto::Room> for (room::Id, Room) {
```

Comment:
I'm not very happy with tuples squating. It seems that this requires more time investment, so we decouple types not into `(room::Id, Room)` tuple, but rather into a comprehensive struct:
```rust
struct Room {
    pub id: room::Id,
    pub spec: room::Spec,
}
```

Reply:
@ilslv it's more about user-facing general types than about types generated from `.proto` files. It seems that having clear separation between non-identifiable`element::Spec` and indentifiable `Element` (as `Id` + `Spec`) is more understandable and obvious.

Reply:
@tyranron I still don't get, where `Element` (`Id` + `Spec`) would be applicable?

```rust
pub struct Member {
    pub spec: Pipeline<endpoint::Id, Endpoint>,
// Doesn't contain `Id` because of   ^^^^^^^^
//                    `.yml` spec.
}
```

Reply:
@ilslv on the user side.

Reply:
@tyranron `Member` is a user-side `struct`, which is used like this:

```rust
api.create(Request { ... }).await?

pub enum Request {
    Member {
        id: member::Id,
        room_id: room::Id,
        member: Box<Member>,
    },
    // ...
}
```

Reply:
@ilslv this thing is confusing:
> ```rust
> pub enum Request {
>     Member {
>         id: member::Id,
>         room_id: room::Id,
>         member: Box<Member>,
>     },
>     // ...
> }
> ```

Having `member: Box<member::Spec>` makes it clearer. Even if we don't need identifiable `Member` type inside this crate.

Reply:
@tyranron `member::Spec` will itself be something like 

```rust
struct Spec {
    id: Id,
    member: Member,
}
```

Or we'll have to duplicate fields, which will make refactoring less intuitive. I don't see how hiding `Id` in 1 place via indentation will make api more intuitive.

Reply:
@ilslv 

> `member::Spec` will itself be something like

What? No.

`Spec` desribes fields of its media element as is not identifiable. Media element reuses its `Spec` for payload plus has an identifier, so is identifiable. No duplication should happen. This idea was originally reflected in YAML specs (and borrowed from [Kubernetes API](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/)).